brush-parser/src/parser.rs

```diff
@@ -254,8 +254,37 @@ peg::parser! {
             bang:bang()? seq:pipe_sequence() { ast::Pipeline { bang: bang.is_some(), seq } }
         rule bang() -> bool = specific_word("!") { true }
 
-        rule pipe_sequence() -> Vec<ast::Command> =
-            c:command() ++ (specific_operator("|") linebreak()) { c }
+        pub(crate) rule pipe_sequence() -> Vec<ast::Command> =
```

Comment:
This logic seems a bit cleaner; thanks for reworking the changes.

It's also getting harder to read and easily understand the `pipe_sequence` rule. Could you please break it up into smaller separable chunks? For example, updating a `Command` with the added redirect seems like something that could be handled in a method on `Command`.

For the parsing of the actual separator operator, that feels worthwhile breaking off into a helper rule (e.g., `pipe_operator()`) that encapsulates the `|` vs. `|&` on line 285.